src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface_kernels.jl

```diff
@@ -282,7 +282,7 @@ end
 
 function split_explicit_free_surface_step!(free_surface::SplitExplicitFreeSurface, model, Δt, χ)
 
-    grid = model.grid
+    grid = free_surface.η.grid
```

Comment:
add a comment here. Are these the same?

Reply:
They are not the same for `Distributed` because the `free_surface` has enlarged halos. I had changed it in #3328 because I thought it was a bug but actually, there was an error cancellation with another bug found in this PR (the radius of the LatitudeLongitudeGrid was not propagated through `with_halo`). 
In this PR I fixed the latter and I saw the correct grid to use is the free surface one

Reply:
You may want to call it `free_surface_grid`.